### PR TITLE
[Merged by Bors] - Update `queryNewlyCreatedClient` to return exact client created

### LINKED ISSUE #1328: Modify `ibc_queryNewlyCreatedClients` rpc to return the exact client that was created
**Suggested Solution**

Modify the rpc method to take the `extrinsic hash` and `blockhash` and return the exact client that was created in that extrinsic.

**Alternatives**

_Describe any alternative solutions or features you considered and why you believe your suggested
solution is preferable._

**Additional Information**

_Provide any additional information that you believe may help us evaluate your suggestion._

### PR DESCRIPTION
**Pull request by Wizdave97**
## Issue
close #1328 

## Description
*Please replace this line with info that is not in the ClickUp ticket, eg: "also bumps dependency foo version"*

## Checklist

- [ ] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_
- [ ] I have updated the `book/` to reflect changes made by this PR _(if applicable)_

### CODE REVIEW

File: frame/ibc/ibc-rpc/src/lib.rs [hunk 1]
```diff
@@ -1354,41 +1378,58 @@ where
...
-				_ => continue,
-			}
-		}
+		let block = self.client.block(&at).ok().flatten().ok_or_else(|| {
+			runtime_error_into_rpc_error("[ibc_rpc]: failed to find block with provided hash")
+		})?;
+		let extrinsics = block.block.extrinsics();
+		let (ext_index, ..) = extrinsics
+			.iter()
+			.enumerate()
```
Comment (blasrodri) [inline frame/ibc/ibc-rpc/src/lib.rs:1394]: Do you need enumerate?

**Review by blasrodri — APPROVED** [review]

File: frame/ibc/ibc-rpc/src/lib.rs [hunk 2]
```diff
@@ -478,9 +483,10 @@ where
 		let para_id = api
 			.para_id(&at)
 			.map_err(|_| runtime_error_into_rpc_error("Error getting para id"))?;
-		// todo: get this, somehow, from pallet_ibc::Config::CHILD_TRIE_KEY
-		// todo: get this, somehow, from pallet_ibc::Config::CHILD_TRIE_KEY
-		let child_info = ChildInfo::new_default(b"ibc/");
+		let child_trie_key = api
+			.child_trie_key(&at)
```
Comment (seunlanlege) [inline frame/ibc/ibc-rpc/src/lib.rs:487]: chef's kiss 🤌🏾

File: frame/ibc/ibc-rpc/src/lib.rs (hunk 1 above)
Comment (seunlanlege) [reply frame/ibc/ibc-rpc/src/lib.rs:1394]: i guess that's how you get the index?

File: runtime/dali/src/lib.rs [hunk 3]
```diff
@@ -1673,21 +1677,31 @@ impl_runtime_apis! {
...
-					},
-					_ => None
-				}
-			});
-
-			events.fold(vec![], |mut events, ev| {
-				events.extend(ev);
-				events
-			})
+		fn block_events(extrinsic_index: Option<u32>) -> Option<Vec<pallet_ibc::events::IbcEvent>> {
```
Comment (seunlanlege) [inline runtime/dali/src/lib.rs:1680]: `Option<Vec<_>>` is a bad pattern, you can just return an empty vec.

**Review by seunlanlege — APPROVED** [review]

File: runtime/dali/src/lib.rs (hunk 3 above)
Comment (Wizdave97) [reply runtime/dali/src/lib.rs:1680]: Yeah, that's how the index is gotten in frame-system, tested it already though and it works.

File: frame/ibc/ibc-rpc/src/lib.rs (hunk 1 above)
Comment (Wizdave97) [reply frame/ibc/ibc-rpc/src/lib.rs:1394]: Yeah, I don't think there's any other way to get the index when searching through the items of a `Vec`

**Review by seunlanlege — APPROVED** [review]

**Review by blasrodri — APPROVED** [review]

**Pull request closed without merging**